omf-bifrost/src/admin/mod.rs

```diff
@@ -0,0 +1,381 @@
+    } else {
+        None
+    }
+}
+
+fn get_drive_on_right(driving_side: &Option<String>) -> Option<i64> {
+    match driving_side.as_deref() {
+        Some("right") => Some(1),
+        Some("left") => Some(0),
+        _ => None,
+    }
+}
+
+fn get_allow_intersection_names(country: &str, admin_config: &AdminConfig) -> i64 {
+    admin_config
+        .allow_intersection_names
+        .get(country)
+        .map(|&b| if b { 1 } else { 0 })
+        .unwrap_or(0)
+}
+
+/// Creates and populates the Valhalla-compatible 'admins' table in SQLite, ingesting and transforming data from the Overture DuckDB source tables.
+fn build_admins_table(
+    duck_con: &DuckConnection,
+    sqlite_con: &Connection,
+    geoparquet_division_path: &str,
+    geoparquet_area_path: &str,
+    admin_config: &AdminConfig,
+) -> Result<()> {
+    info!("Creating admins table");
+    sqlite_con.execute_batch(CREATE_ADMINS_SQL)?;
+
+    info!("Adding geometry column to admins table");
+    sqlite_con.execute_batch(ADD_ADMINS_GEOM_SQL)?;
+
+    info!("Querying and joining division and area tables in DuckDB");
+    let select_query = format!(
+        "WITH divs AS (
+            SELECT
+                id as div_id,
+                parent_division_id,
+                norms.driving_side
+         FROM read_parquet('{}')
+        )
+        SELECT
+            area.division_id,
+            divs.parent_division_id,
+            CASE area.subtype
+                WHEN 'country' THEN 2
+                WHEN 'dependency' THEN 2
+                WHEN 'region' THEN 4
+                ELSE NULL END as admin_level,
+            area.country,
+            area.region,
+            area.names.primary as name,
+            area.names.common.en as name_en,
+            divs.driving_side,
+            ST_AsText(area.geometry) as wkt
+        FROM read_parquet('{}') as area
+        JOIN divs ON area.division_id = divs.div_id
+        WHERE area.is_land = TRUE
+            AND area.geometry IS NOT NULL
+            AND area.subtype IN ('country','dependency','region')",
+        geoparquet_division_path, geoparquet_area_path
+    );
+
+    let mut stmt = duck_con.prepare(&select_query)?;
+    let mut rows = stmt.query([])?;
+
+    info!("Processing admin records");
+    sqlite_con.execute_batch("BEGIN;")?;
+    let mut admin_count = 0u64;
+    while let Some(row) = rows.next()? {
+        let division_id: String = row.get(0)?;
+        let parent_division_id: Option<String> = row.get(1)?;
+        let admin_level: Option<i64> = row.get(2)?;
+        let country: String = row.get(3)?;
+        let region: Option<String> = row.get(4)?;
+        let name: Option<String> = row.get(5)?;
+        let name_en: Option<String> = row.get(6)?;
+        let driving_side: Option<String> = row.get(7)?;
+        let geom_wkt: String = row.get(8)?;
+
+        let admin_level = match admin_level {
+            Some(lvl) => lvl,
+            _ => continue,
+        };
```

Comment:
Are we expecting to always find an `admin_level` per row here? What happens if we don't find any `admin_level`? Should this fail?